core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2809,12 +2814,19 @@ wasmtime_ssp_sock_recv_from(wasm_exec_env_t exec_env, struct fd_table *curfds,
     __wasi_errno_t error;
     bh_sockaddr_t sockaddr;
     int ret;
+    /* make a copy because the value pointed seem to be modifed */
```

Comment:
modifed -> modified

Do you mean src_addr will be modified below? But it is only used in the end of function: 
`bh_sockaddr_to_wasi_addr(&sockaddr, src_addr)`

And you convert src_addr_copy to sockaddr with `wasi_addr_to_bh_sockaddr(&src_addr_copy, &sockaddr)`, and then pass `sockaddr` to blocking_op_socket_recv_from, but it seems that the input fields of `sockaddr` are not used in posix os_socket_recv_from:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/shared/platform/common/posix/posix_socket.c#L258-L284

I think it is because that in your Zephyr's os_socket_recv_from implementation, you need the input value of sockaddr?
```C
temp_addr->sa_family = src_addr->is_ipv4 == true ? AF_INET : AF_INET6;
```
If yes, then had better change the comment to make it more accurate.

Reply:
Not sure why I made this copy, restested in both case with and without copy and the original implementation worked.